### PR TITLE
separate build with python3 then run test with python2 

### PR DESCRIPTION
**Pull request by alakendu**
This PR is top of https://github.com/open-traffic-generator/openapiart/pull/269 as well as sync with main and file change to Python3

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -1,58 +1,74 @@
...
         uses: actions/checkout@v2
         with:
           ref: ${{ github.head_ref }}
-      - name: Setup Go for CI
-        uses: actions/setup-go@v2
+      - uses: actions/setup-go@v2
         with:
-          go-version: '1.17.3'
-      - name: Setup Go and protoc
+          go-version: "^1.17.1"
```
Comment (ashutshkumr) [inline .github/workflows/build.yml:26]: let's use a fixed version of go (1.17.5)

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -1,58 +1,74 @@
...
+      - name: Setup protoc
         run: |
-          python do.py setup_ext
+          python do.py get_protoc
```
Comment (ashutshkumr) [inline .github/workflows/build.yml:29]: use `python setup_ext` only (since it checks for incorrect platform)

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -61,28 +76,54 @@ jobs:
...
+      - name: Display structure of downloaded files
+        run: ls -R
+      - uses: actions/setup-go@v2
+        with:
+          go-version: "1.17.5"
+      - name: Setup protoc
         run: |
-          python do.py go_lint
-      - name: Run tests
+          python do.py setup_ext
```
Comment (ashutshkumr) [inline .github/workflows/build.yml:106]: Do we really need to do this ?

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -61,28 +76,54 @@ jobs:
...
+        uses: actions/checkout@v2
         with:
-          name: generated-artifacts
-          path: art
+          ref: ${{ github.head_ref }}
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+      - name: Generate only for 3.9
```
Comment (ashutshkumr) [inline .github/workflows/build.yml:123]: can't we just download artifacts and post them (since that's the one we've tested) ?

File: do.py [hunk 5]
```diff
@@ -132,16 +132,17 @@ def generate():
     )
 
 
-def test():
+def test_python():
```
Comment (ashutshkumr) [inline do.py:135]: i would suggest shortening the name to `testpy` and `testgo`.

File: .github/workflows/build.yml (hunk 1 above)
Comment (alakendu) [reply .github/workflows/build.yml]: set to 1.17.5

File: .github/workflows/build.yml (hunk 2 above)
Comment (alakendu) [reply .github/workflows/build.yml]: change setup_ext

File: .github/workflows/build.yml (hunk 3 above)
Comment (alakendu) [reply .github/workflows/build.yml]: agrre

**Review by ashutshkumr — APPROVED** [review]

**Pull request merged**